src/uwtools/scheduler.py

```diff
@@ -269,33 +272,28 @@ class LSF(JobScheduler):
         RequiredAttribs.QUEUE: "-q",
         RequiredAttribs.ACCOUNT: "-P",
         RequiredAttribs.WALLTIME: "-W",
-        RequiredAttribs.NODES: "-n",
+        RequiredAttribs.NODES: lambda x: f"-n {x}",
         RequiredAttribs.TASKS_PER_NODE: lambda x: f"-R span[ptile={x}]",
         OptionalAttribs.SHELL: "-L",
         OptionalAttribs.JOB_NAME: "-J",
         OptionalAttribs.STDOUT: "-o",
         OptionalAttribs.THREADS: lambda x: f"-R affinity[core({x})]",
+        OptionalAttribs.MEMORY: lambda x: f"-R rusage[mem={x}]",
     }
 
-    def pre_process(self):
+    def pre_process(self) -> Dict[str, Any]:
         items = self._data
-        items[self._map[OptionalAttribs.THREADS](items[OptionalAttribs.THREADS])] = ""
+        # LSF requires threads to be set (if None is provided, default to 1)
+        items[OptionalAttribs.THREADS] = items.get(OptionalAttribs.THREADS, 1)
+        memory = items.get(OptionalAttribs.MEMORY, "")
+        if memory not in NONEISH:
+            items[self._map[OptionalAttribs.MEMORY](items[OptionalAttribs.MEMORY])] = ""
```

Comment:
I'm confused by the logic here. I read it as:
- Get memory
- If memory is set to a value, save it to the config as an empty string

Is that right? Is it all about the key in this case because we're filling in the value with a `lambda` function?

Also, do you need to interrogate `items[OptionalAttribs.MEMORY]` again, or can you just reference `memory` where it's already stored?

Reply:
You got the 1st part right:
1. Get memory.

If it was provided, the `items` dictionary now contains a new element with the map of what that memory line translates to for the chosen scheduler, the value of which is an empty string.
e.g. in this case (LSF), the new element in `items` will be:
```
items[-R rusage[mem=1MB] = ""
```

Reply:
It took me a minute to realize that in this "special" case, we'd be returning a lambda function and passing it the input value of memory. And that once you do that, the key will be the value for this item (thanks to a brush up on this data structure from @ryanlong1004).